contrib/Dockerfile.test

```diff
@@ -34,7 +34,7 @@
 #   docker run --privileged --group-add keep-groups -v ./critest_exit_code.txt:/tmp/critest_exit_code.txt containerd-test
 # ------------------------------------------------------------------------------
 
-ARG GOLANG_VERSION=1.24.1
+ARG GOLANG_VERSION=1.24.2
```

Comment:
Can you make this one 1.23.8? Historically, this was the best location for our pipelines to get the version to use to build, and I think our script still greps this file for `GOLANG_VERSION`

Reply:
Curious, Isnt the best place to get the version here https://github.com/containerd/containerd/blob/main/.github/actions/install-go/action.yml#L6.

Reply:
Have updated here also.